### PR TITLE
feat(CI): add MacOS CI testing

### LINKED ISSUE #85: CI tests are not being run on MacOS
CI tests are having SAM / Docker issues on MacOS

### PR DESCRIPTION
**Pull request by mmaeng**
Closes #85 

Run CI testing on MacOS to help ensure no breaking changes.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mmaeng** [comment]
SAM CLI v.1.47 is needed to support colima (alternative to docker engine, which cannot be used because of licensing) on MacOS.  SAM v1.47.0 is not compatible with Python 3.6 and Python 3.6 was removed from the testing suite.

**Comment by mmaeng** [comment]
MacOS - docker and colima on Github Runners
https://github.com/actions/runner-images/issues/17

SAM Colima Support v1.47.0
https://github.com/aws/aws-sam-cli/issues/3595

SAM deprecate Python 3.6 in v1.42
https://github.com/aws/aws-sam-cli/releases/tag/v1.42.0

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,8 +26,39 @@ jobs:
...
+        run: |
+          brew install docker
+          # Docker engine is no longer available because of licensing
+          # Alternative Colima is part of the github macOS runner
+          # SAM v1.47.0+ needed for colima support, unable to use Python 3.6
+          colima start
+          # Ensure colima is configured for later user
+          echo "DOCKER_HOST=unix://$HOME/.colima/default/docker.sock" >> $GITHUB_ENV
+          # Verify Docker
+          docker ps
```
Comment (kddejong) [inline .github/workflows/ci.yml:58]: Hows this verify docker?  Seems like extra and for logging purposes?

Comment (mmaeng) [reply .github/workflows/ci.yml:58]: I wanted to do a docker CLI to server connection check.
`docker info`
`docker version`

would also be options that fail if docker CLI cannot connect with the backend.  Though those are more verbose.

**Review by kddejong — APPROVED** [review]

**Review by ericzbeard — APPROVED** [review]

**Pull request merged**